### ISSUE TITLE
Feature to name container from "docker-compose run"

### ISSUE THREAD

**Issue by EntilZha** [docker/compose#1347]
This is a feature proposal/request to add a capability to `docker-compose run`. I have checked in the docker and docker-dev IRC channels and looked at prior issues and it seems this isn't on the radar.

Currently, if you have a `docker-compose.yml` you can start a specific container with possibly its dependencies using `docker-compose run CONTAINER_NAME`. However, this starts/creates a new container with the ordinary compose name appended by `_run_1`, where `1` could be any number. There is no way to tell `docker-compose` what this should be named.

The main use case would be overriding the run command for a specific container (in my case flags for development/production mode), but keeping the original name. This change would allow for this and also be flexible enough to support any name.

**Comment by dnephin** [comment]
This is blocked on #1066, and there is discussion there about making this possible once it's complete.

**Comment by EntilZha** [comment]
Sounds great that its on the roadmap.

**Comment by aanand** [comment]
Closing in favour of https://github.com/docker/compose/issues/1518

**Comment by jylin** [comment]
Thanks for this feature. I notice that "docker-compose run <target>" doesn't seem to use the custom name. I think I would expect that as well to use the container_name as opposed to just for "up".

**Comment by xelibrion** [comment]
@aanand I just experienced the same issue as @jylin - `docker-compose run` does not seem to use `container_name` configuration parameter, seems like a bug.

**Comment by alkiskal** [comment]
Any news on what @xelibrion mentions? I experienced the exact same behaviour `docker-compose version 1.9.0, build 2585387`
